### DISCUSSION TITLE
Vercel's build cache makes images low res on every deploy with Hugo's image processing

### DISCUSSION THREAD

**Discussion by adityatelange** [vercel/vercel#5424] [Help]
I am using Hugo to deploy my personal website/blog at https://adityatelange.me.
I use SSG `hugo`([theme PaperMod](https://github.com/adityatelange/hugo-PaperMod/)) to build and deploy it with `vercel`, using a private git repo, thanks to vercel's automatic builds on push.

Build process includes automatic `srcset` generation, which generates images for various sizes optimizing it for better performance, on smaller screens as well as large displays.

However on each build, because of vercel's build cache previous resources are synced, and every-time I deploy it the images of lower resolution are generated, gradually decreasing the image quality. :(

What could be possible solution to this ?

**Comment by RickCogley** [comment]
This thread is old but, I just started using vercel and was curious about this so I tested it (edit: not using papermod theme, just manually).

After a change in the text and a subsequent deployment triggered by a commit and push to main, I did _not_ find any visual difference, and my resource names in the `srcset` were the same. 

In project settings, I've got Hugo selected as the framework, and I've left the build command as default. 

So it could have been the way vercel was building hugo by default, the hugo version itself or your build command?

**Comment by RickCogley** [comment]
Right, now I specified the build step in the vercel project settings, using a typical set of switches for me: 

```
echo "=== build it ===" && env && hugo --gc --minify --ignoreCache --verbose
```

Same result as above, except now the HTML is minified. Otherwise no bad impact.
